components-api/src/main/scala/pl/touk/nussknacker/engine/api/Context.scala

```diff
@@ -85,7 +100,13 @@ object ContextId {
 case class Context(
     id: ContextId,
     variables: Map[String, Any],
-    parentContext: Option[Context]
+    parentContext: Option[Context],
+    /**
+     * Optional internal tracking ID used for debugging, monitoring, or tracing the execution flow.
+     * For developer use only – not exposed to process designers or end users.
+     * Helps propagate correlation IDs across system components.
+     */
+    traceId: Option[TraceId]
```

Comment:
Consider options:
- traceId - is more from OpenTelemetry
- trackingId - more business
- correlationId - more custom

Reply:
If it is a vote I think that traceId has a good sound in it - I would keep that :+1:

Reply:
Let's improve the comment a bit:
1. "For developer use only" - This feature is not exactly for developers. A user can see the tracing information in messages passed to services. I think that we can skip this sentence.
2. "not exposed to process designers or end users" - 
2.1. process -> scenario. 
2.2. designers - do you mean the Designer application? or people that design scenarios? If 1) I would replace it with a singular noun and remove "process/scenario"  
2.3. Let's add "For now" - I think that the current state is temporal, we would like to expose this information in expressions